addons/account/wizard/setup_wizards.py

```diff
@@ -27,26 +27,37 @@ def _compute_opening_move_posted(self):
 
     @api.constrains('fiscalyear_last_day', 'fiscalyear_last_month')
     def _check_fiscalyear(self):
-        # We try if the date exists in 2020, which is a leap year.
-        # We do not define the constrain on res.company, since the recomputation of the related
-        # fields is done one field at a time.
         for wiz in self:
             try:
-                date(2020, int(wiz.fiscalyear_last_month), wiz.fiscalyear_last_day)
+                year, fiscalyear_last_day = self._get_fiscalyear_and_last_day()
+                date(year, int(wiz.fiscalyear_last_month), fiscalyear_last_day)
             except ValueError:
                 raise ValidationError(
                     _('Incorrect fiscal year date: day is out of range for month. Month: %s; Day: %s') %
-                    (wiz.fiscalyear_last_month, wiz.fiscalyear_last_day)
+                    (wiz.fiscalyear_last_month, fiscalyear_last_day)
                 )
 
+    def _get_fiscalyear_and_last_day(self):
+        if self.opening_date:
+            year = self.opening_date.year
+        else:
+            year = datetime.now().year
+        if self.company_id.fiscalyear_last_day == 29 and self.company_id.fiscalyear_last_month == '2':
+            fiscalyear_last_day = self.company_id._get_fiscalyear_february_last_day(year)
+        else:
+            fiscalyear_last_day = self.company_id.fiscalyear_last_day
```

Comment:
I don't understand why you're factorizing with the function `_get_fiscalyear_february_last_day`, but in fact, use it in the same way thrice, with this same block of code.
Why not create a function `get_fiscalyear_last_day` as proposed by wan, that gets you the day and the month of the fiscal year based on the year, whether it's the 29/02 or not, so you don't have to repeat the logic?